Спринт №9

Keithcutt opened the pull request:
Надеюсь в этот раз проект соберется. Вроде все файлы на месте

Keithcutt committed:
Домашнее задание 9 спринта:

Тема I. Стили и темы
1. Повторябщиеся атрибуты у однотипных вью вынесены в стили.
2. Для всех экранов добавлена поддержка темной темы.
3. Убраны тени у кнопок.
4. Подключены необходимые шрифты для текстов на экране.

Тема II. Context и Intent
1. Все строки с экранов перенесены в ресурсы.
2. Реализованы кнопки «Поделиться приложением», «Написать разработчикам», «Пользовательское соглашение».

Тема III. EditText и TextWatcher
1. Реализован экран «Поиск» без выполнения поискового запроса и отображения списка результатов.

Тема IV. Хранение данных. Начало
1. Реализовано сохранение текста поискового запроса в жизненном цикле SearchActivity.

Keithcutt committed:
Домашнее задание 9 спринта:

Тема I. Стили и темы
1. Повторябщиеся атрибуты у однотипных вью вынесены в стили.
2. Для всех экранов добавлена поддержка темной темы.
3. Убраны тени у кнопок.
4. Подключены необходимые шрифты для текстов на экране.

Тема II. Context и Intent
1. Все строки с экранов перенесены в ресурсы.
2. Реализованы кнопки «Поделиться приложением», «Написать разработчикам», «Пользовательское соглашение».

Тема III. EditText и TextWatcher
1. Реализован экран «Поиск» без выполнения поискового запроса и отображения списка результатов.

Тема IV. Хранение данных. Начало
1. Реализовано сохранение текста поискового запроса в жизненном цикле SearchActivity.

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,77 @@
+        return if (s.isNullOrEmpty()) {
+            View.GONE
+        } else {
+            View.VISIBLE
```
Для изменения видимости элемента на экране в данном случае удобнее использовать extension View.isVisible:
View.isVisible = true соответствует View.visibility = View.VISIBLE
View.isVisible = false соответствует View.visibility = View.GONE

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,77 @@
+
+    companion object {
+        const val TEXT_INPUT = "TEXT_INPUT"
+        const val TEXT_DEF = ""
```
private

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,77 @@
+        const val TEXT_DEF = ""
+    }
+
+    private var savedInput: String? = null
```
Принято объявлять переменные класса в начале класса, а не в конце

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -14,5 +16,34 @@ class SettingsActivity : AppCompatActivity() {
+                putExtra(Intent.EXTRA_SUBJECT, getString(R.string.email_subject))
+                putExtra(Intent.EXTRA_TEXT, getString(R.string.email_text))
+            }
+            startActivity(supportIntent)
```
Можно упростить до
```
Intent().apply {
    action = Intent.ACTION_SENDTO
    data = Uri.parse("mailto:")
    putExtra(Intent.EXTRA_EMAIL, arrayListOf(getString(R.string.email_recipient)))
    putExtra(Intent.EXTRA_SUBJECT, getString(R.string.email_subject))
    putExtra(Intent.EXTRA_TEXT, getString(R.string.email_text))
    startActivity(this)
}
```

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,56 @@
+    </style>
+
+    <style name="SettingsItemsIcons">
+        <item name="android:layout_gravity">end</item>
```
Лучше внешние отступы, высоту и ширину, веса, внешние гравити не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи и вообще только за то, что находится внутри нее - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

Keithcutt committed:
Домашнее задание 9 спринта:

Тема I. Стили и темы
1. Повторябщиеся атрибуты у однотипных вью вынесены в стили.
2. Для всех экранов добавлена поддержка темной темы.
3. Убраны тени у кнопок.
4. Подключены необходимые шрифты для текстов на экране.

Тема II. Context и Intent
1. Все строки с экранов перенесены в ресурсы.
2. Реализованы кнопки «Поделиться приложением», «Написать разработчикам», «Пользовательское соглашение».

Тема III. EditText и TextWatcher
1. Реализован экран «Поиск» без выполнения поискового запроса и отображения списка результатов.

Тема IV. Хранение данных. Начало
1. Реализовано сохранение текста поискового запроса в жизненном цикле SearchActivity.

Keithcutt committed:
Исправления после первого ревью
